src/SHA256.mo

```diff
@@ -134,17 +134,18 @@ module {
     };
 
     private func block(data : [Nat8]) {
-      var p = data;
       var w = Array.init<Nat32>(64, 0);
-      while (p.size() >= 64) {
+      var start = 0;
+      var len = data.size();
+      while (len >= 64) {
         var j = 0;
         for (i in Iter.range(0, 15)) {
           j := i * 4;
           w[i] :=
-            Nat32.fromIntWrap(Nat8.toNat(p[j + 0])) << 24 |
-            Nat32.fromIntWrap(Nat8.toNat(p[j + 1])) << 16 |
-            Nat32.fromIntWrap(Nat8.toNat(p[j + 2])) << 08 |
-            Nat32.fromIntWrap(Nat8.toNat(p[j + 3])) << 00;
+            Nat32.fromIntWrap(Nat8.toNat(data[start + j + 0])) << 24 |
```

Comment:
Can't these just be Nat32.fromNat or do you want to avoid the redundant overflow check?

Reply:
You mean Nat32.fromNat(Nat8.toNat(..))?

Reply:
Yup. But perhaps the existing is faster since it doesn't check for overflow

Reply:
I tested it. fromIntWrap is slightly cheaper than fromNat.